### ISSUE TITLE
we no longer show history if you're kicked from a room?

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-meta#465]
so you can't see why you were kicked, or access your history

**Comment by allo-** [comment]
What about being able to delete messages from the room? This is GDPR relevant (#820) as even a kicked user has GDPR rights on the message history.

**Comment by FSG-Cat** [comment]
Well i seem to have run into a bug like this or this exact bug but with bans. On Element Nightly Desktop and this has been a thing for a similar amount of time to the age of this issue as far as my memory recalls. 

> What about being able to delete messages from the room? This is GDPR relevant (#820) as even a kicked user has GDPR rights on the message history.

@allo- that is Spec blocked not Element blocked as far as i am aware when it comes to non members sending events.

**Comment by allo-** [comment]
@FSG-Cat What does that mean? I came here clicking around through the different bugs concerning if matrix can delete rooms by now, but when I read the issue, I thought that kicking usually would not delete message history, but a user must be able to delete it (and therefore at least see the old messages to be able to find what to delete) for the server to be GDPR compliant.

**Comment by FSG-Cat** [comment]
@allo- the problem is that your redaction events will fail event authorisation if your not in the room so there is that small issue. Like there are a lot of moving parts and potential ways to fix it and tbh UX is one of the issues that has to be solved.

**Comment by allo-** [comment]
Yes, that's what I am meaning. Either a kick (or only a ban) would need to remove all content instantly, or still allow redaction and correction, which are GDPR rights.
